UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.Common/Helpers/AzureEventGridDomainService.cs

```diff
@@ -18,25 +18,58 @@ namespace UKHO.ExternalNotificationService.Common.Helpers
             
             EventGridManagementClient eventGridMgmtClient = await GetEventGridClient(_eventGridDomainConfig.SubscriptionId, cancellationToken);           
             DomainTopic topic = await GetDomainTopic(eventGridMgmtClient, subscriptionRequestMessage.NotificationTypeTopicName, cancellationToken);
+            string eventSubscriptionScope = topic.Id;
+
+            string deadLetterDestinationResourceId = $"/subscriptions/{_eventGridDomainConfig.SubscriptionId}/resourceGroups/{_eventGridDomainConfig.ResourceGroup}/providers/Microsoft.Storage/storageAccounts/{_subscriptionStorageConfiguration.StorageAccountName}";
+
+            EventSubscription eventSubscription = new() {
+                Destination = new WebHookEventSubscriptionDestination()
+                {
+                    EndpointUrl = subscriptionRequestMessage.WebhookUrl
+                },
+                // The below are all optional settings
+                EventDeliverySchema = EventDeliverySchema.CloudEventSchemaV10,
+                /* Retry policy decides when an event can be marked as expired. 
+                   The default retry policy keeps the event alive for 24 hrs (=1440 mins or 30 retries with exponential backoffs)
+                   An event is marked as expired once any of the retry policy limits are exceeded. 
+                   Note: The below configuration for the retry policy will cause events to expire after one delivery attempt. 
+                   This is only to make it easier to help test/verify dead letter destinations quickly.
+                */
+                RetryPolicy = new RetryPolicy()
+                {
+                    MaxDeliveryAttempts = _eventGridDomainConfig.MaxDeliveryAttempts,
+                    EventTimeToLiveInMinutes = _eventGridDomainConfig.EventTimeToLiveInMinutes,
+                },
+                // With dead-letter destination configured, all expired events will be delivered to this destination.
+                // Note: only Storage Blobs are supported as dead letter destinations as of now.
+                DeadLetterDestination = new StorageBlobDeadLetterDestination()
+                {
+                    ResourceId = deadLetterDestinationResourceId,
+                    BlobContainerName = _subscriptionStorageConfiguration.StorageContainerName,
+                }
+            };
+            EventSubscription createdOrUpdatedEventSubscription = await eventGridMgmtClient.EventSubscriptions.CreateOrUpdateAsync(eventSubscriptionScope, subscriptionRequestMessage.SubscriptionId, eventSubscription, CancellationToken.None);
 
```

Comment:
Please pass the cancellationToken instead of  CancellationToken.None

Reply:
added